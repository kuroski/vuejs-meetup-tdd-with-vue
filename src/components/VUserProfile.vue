<script>
export default {
  name: 'user-profile',
  props: {
    user: {
      type: Object,
      required: true,
    }
  }
}
</script>

<template>
  <div>
    <img :src="user.avatar_url" />
    <div class="name">
      {{ user.name }}
    </div>
  </div>
</template>

