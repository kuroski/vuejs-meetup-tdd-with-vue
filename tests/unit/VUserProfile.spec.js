import { shallowMount } from '@vue/test-utils'
import VUserProfile from '@/components/VUserProfile'
import user from './fixtures/user'

describe('VUserProfile', () => {
  let props

  const build = () => {
    const wrapper = shallowMount(VUserProfile, {
      propsData: props,
    })
    return {
      wrapper,
      avatar: () => wrapper.find('.user-profile__avatar'),
      name: () => wrapper.find('.user-profile__name')
    }
  }

  beforeEach(() => {
    props = {
      user: user.success
    }
  })

  it('renders the component', () => {
    // arranje
    const { wrapper } = build()
    // assert
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders main components', () => {
    // arranje
    const { avatar, name } = build()
    // assert
    expect(avatar().exists()).toBe(true)
    expect(avatar().attributes().src).toBe(props.user.avatar_url)
    expect(name().exists()).toBe(true)
    expect(name().text()).toBe(props.user.name)
  })
})