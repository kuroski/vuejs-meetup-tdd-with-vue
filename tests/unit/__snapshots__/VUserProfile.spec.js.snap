// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VUserProfile renders the component 1`] = `
<div class="user-profile">
  <img src="https://github.com/images/error/octocat_happy.gif" class="user-profile__avatar">
  <div class="user-profile__name">
    monalisa octocat
  </div>
  <div class="user-profile__bio">
    There once was...
  </div>
</div>
`;
